### PR TITLE
Added --lock option for "poetry remove"

### PR DESCRIPTION
**Pull request by agronholm**
# Pull Request Check List

Resolves: #3971

- [ ] Added **tests** for changed code.
- [X] Updated **documentation** for changed code.

Poetry already supports "add --lock" and "update --lock" but for some reason the corresponding "remove --lock" feature was missing.

### CODE REVIEW

**Comment by agronholm** [comment]
I need some help writing a test for this.

**Comment by agronholm** [comment]
Also next I might want to address the fact that these commands still create a useless virtualenv.

File: poetry/console/commands/remove.py [hunk 1]
```diff
@@ -73,6 +74,8 @@ def handle(self) -> int:
         self._installer.verbose(self._io.is_verbose())
         self._installer.update(True)
         self._installer.whitelist(requirements)
+        if self.option("lock"):
+            self._installer.lock()
 
         status = self._installer.run()
```
Comment (abn) [inline poetry/console/commands/remove.py:80]: Intaller should not run if its a lock only op.

**Comment by agronholm** [comment]
Why does the `add` command then do the same? I wouldn't know what to do in this situation which is why I asked for help on Discord.

File: poetry/console/commands/remove.py (hunk 1 above)
Comment (agronholm) [reply poetry/console/commands/remove.py:80]: Why does the add command then do the same? I wouldn't know what to do in this situation which is why I asked for help on Discord.

**Comment by neersighted** [comment]
I'm going to close this as it seems to be stalled out and very naive in its current form. I would be happy to review a future version with robust tests.

**Comment by agronholm** [comment]
It's stalled because I never got any help with the implementation, particularly with tests. I would be happy to continue work on this if somebody gave me the information I'm missing.

**Comment by neersighted** [comment]
I'd take a look at similar tests around locking -- particularly `test_lock.py`. It's a pretty good example of the level of mocking needed to set up a command-line test, as well as how to invoke commands and check Poetry's results after.

**Pull request closed without merging**